### PR TITLE
Integration with Mongoose 8.4 listDatabases()

### PR DESCRIPTION
**Pull request by vkarpov15**
**What this PR does**:

Mongoose 8.4 is introducing a `listDatabases()` function, which lines up nicely with `findNamespaces()`. Once Mongoose 8.4 is released, we can ship this. This change should be backwards compatible with older versions of Mongoose too.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/driver/connection.ts [hunk 1]
```diff
@@ -93,6 +91,14 @@ export class Connection extends MongooseConnection {
         });
     }
 
+    async listDatabases(): Promise<{ databases: string[] }> {
+        return executeOperation(async () => {
+            await this._waitForClient();
+            const { status } = await this.client.findNamespaces();
```
Comment (kathirsvn) [inline src/driver/connection.ts:97]: If the `response` from the Data API has `errors`, the `executeCommandWithUrl` function throws `StargateServerError`. Do we need a test case for that to check if the error is handled gracefully?

Comment (vkarpov15) [reply src/driver/connection.ts:97]: @kathirsvn is there a way to get `findNamespaces()` to return a server error? Based on the [implementation](https://github.com/stargate/data-api/blob/0c3859a5e9e4605c42b1f75610842ee74e23dfcf/src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindNamespacesOperations.java#L21) and API I don't see a good way to get `findNamespaces()` to error out. I could sinon.stub(), but that seems a bit contrived.

Comment (kathirsvn) [reply src/driver/connection.ts:97]: Got it, it is not thrown by the API, functionally for any input. Even Invalid token returns 401 and not the error body I think.

**Review by vkarpov15 — COMMENTED** [review]
@kathirsvn Mongoose 8.4 is out, so this PR should be good to go.

With the way stargate-mongoose is implemented, we don't need to drop support for Mongoose versions before 8.4. `listDatabases()` will work as expected even in the older versions.

**Review by kathirsvn — APPROVED** [review]

**Pull request merged**